consensus/types/src/beacon_state.rs

```diff
@@ -2277,13 +2336,27 @@ impl<E: EthSpec> BeaconState<E> {
             consolidation_balance_to_consume
                 .safe_add_assign(additional_epochs.safe_mul(per_epoch_consolidation_churn)?)?;
         }
-        // Consume the balance and update state variables
-        let state = self.as_electra_mut()?;
-        state.consolidation_balance_to_consume =
-            consolidation_balance_to_consume.safe_sub(consolidation_balance)?;
-        state.earliest_consolidation_epoch = earliest_consolidation_epoch;
-
-        Ok(state.earliest_consolidation_epoch)
+        match self {
+            BeaconState::Electra(_) => {
+                let state = self.as_electra_mut()?;
+
+                // Consume the balance and update state variables.
+                state.consolidation_balance_to_consume =
+                    consolidation_balance_to_consume.safe_sub(consolidation_balance)?;
+                state.earliest_consolidation_epoch = earliest_consolidation_epoch;
+                Ok(state.earliest_consolidation_epoch)
+            }
+            BeaconState::Fulu(_) => {
+                let state = self.as_fulu_mut()?;
+
+                // Consume the balance and update state variables.
+                state.consolidation_balance_to_consume =
+                    consolidation_balance_to_consume.safe_sub(consolidation_balance)?;
+                state.earliest_consolidation_epoch = earliest_consolidation_epoch;
+                Ok(state.earliest_consolidation_epoch)
+            }
+            _ => unreachable!(),
```

Comment:
This will panic in the future if we forget to add the G fork here, and compiler wouldn't be able to catch. May be worth returning an `Error`?

Reply:
Ah good catch, I had intended to remove this

Reply:
I have also now added an exhaustive match statement rather than using a default. This seems like the kind of use case we want the compiler to direct us to.